cloudwatch_logs_common/src/log_publisher.cpp

```diff
@@ -112,6 +112,7 @@ void LogPublisher::Run()
 {
   Aws::CloudWatchLogs::ROSCloudWatchLogsErrors publisher_status = CW_LOGS_SUCCEEDED;
   Aws::String next_token;
+  this->cloudwatch_facade_->GetLogStreamToken(this->log_group_, this->log_stream_, next_token);
   while (this->thread_keep_running_.load(std::memory_order_acquire)) {
     // if log group cannot be created or log stream doesnt already exist, don't try to send to CW
     if (!this->does_group_exist_) {
```

Comment:
I'm not sure why the group/stream checks (and creation if needed) are part of the loop: once we set `does_group_exist = true`, we'll never set it back to `false` - so we're not covering the case in which the resources were deleted during the run. 
So unless I'm missing something, it'd be nicer to move all of that logic out of the loop, as well as the call to `GetLogStreamToken`, but keep the current order so that `GetLogStreamToken` gets called after we verified that the stream exists (that way we avoid the red-herring log messages that Tim mentioned).

Reply:
I'm refactoring it slightly into a state machine to be cleaner. These will still be in the main execution loop because we're using that as a retry mechanism in the case when the calls are failing to create resources.